metaflow/event_logger.py

```diff
@@ -1,33 +1,31 @@
-from .sidecar import SidecarSubProcess
-from .sidecar_messages import Message, MessageTypes
+from metaflow.sidecar import Message, MessageTypes, Sidecar
 
 
-class NullEventLogger(object):
-    def __init__(self, *args, **kwargs):
-        pass
+class BaseEventLogger(object):
+    TYPE = "nullSidecarLogger"
 
-    def start(self):
-        pass
+    def __init__(self, flow=None, env=None):
+        self._sidecar = Sidecar(self.TYPE)
 
-    def log(self, payload):
-        pass
+    def start(self):
+        return self._sidecar.start()
 
     def terminate(self):
-        pass
+        return self._sidecar.terminate()
 
+    def add_to_context(self, **kwargs):
+        pass
 
-class EventLogger(NullEventLogger):
-    def __init__(self, logger_type):
-        # type: (str) -> None
-        self.sidecar_process = None
-        self.logger_type = logger_type
+    def log(self, payload):
```

Comment:
we can combine `log` and `add_to_context` into `log(self, message)`

Reply:
As discussed, I kept log as is and added send(self, msg). Also added send(self, msg) to the monitor.